File: csharp/squeaky-clean/SqueakyClean.cs
using System;
using System.Text;

public static class Identifier
{
    public static string Clean(string identifier)
    {
        StringBuilder sb = new StringBuilder();
        bool prevDash = false;

        foreach (char c in identifier)
        {
            if (c == ' ')
            {
                sb.Append('_');
            }
            else if (Char.IsControl(c))
            {
                sb.Append("CTRL");
            }
            else if (c == '-')
            {
                prevDash = true;
            }
            else if (Char.IsLetter(c) && !(c >= 'α' && c <= 'ω'))
            {
                sb.Append(prevDash ? Char.ToUpper(c) : c);
                prevDash = false;
            }
        }
        return sb.ToString();
    }
}

File: java/squeaky-clean/src/main/java/SqueakyClean.java
import java.util.HashMap;
import java.util.Map;

class SqueakyClean {
    static String clean(String identifier) {
        StringBuilder sBuilder = new StringBuilder();
        boolean kebapCase = false;
        
        for (int index = 0; index < identifier.length(); index++) {
            char currentChar = identifier.charAt(index);
            char modifiedChar;
            if (currentChar == '-') {
                kebapCase = true;
                continue;
            }
            
            if (currentChar == ' ') {
                modifiedChar = '_';
            } else if (currentChar == '0') {
                modifiedChar = 'o';
            } else if (currentChar == '1') {
                modifiedChar = 'l';
            } else if (currentChar == '3') {
                modifiedChar = 'e';
            } else if (currentChar == '4') {
                modifiedChar = 'a';
            } else if (currentChar == '7') {
                modifiedChar = 't';
            } else if (!Character.isLetter(currentChar)) {
                continue;
            } else {
                modifiedChar = currentChar;
            }

            if (kebapCase) {
                modifiedChar = Character.toUpperCase(modifiedChar);
                kebapCase = false;
            }
            sBuilder.append(modifiedChar);
        }

        return sBuilder.toString();
    }
}
